### PR TITLE
Add total_on_hand for variants api response

### PR DESCRIPTION
**Pull request by medius**

### CODE REVIEW

**Comment by IrvinFan** [comment]
Reviewing

File: api/spec/controllers/spree/api/products_controller_spec.rb [hunk 1]
```diff
@@ -401,6 +401,32 @@ module Spree
           json_response["errors"]["name"].should == ["can't be blank"]
         end
 
+        context 'with existing variants' do
+          before do
+            variant_hash = {
+              :sku => '123', :price => 19.99, :options => [{:name => "size", :value => "small"}]
+            }
+            @variant_id = product.variants.create!({ product: product }.merge(variant_hash)).id
```
Comment (IrvinFan) [inline api/spec/controllers/spree/api/products_controller_spec.rb]: We can leverage the factory to create the variant. Something like `@variant_id = create(:variant, product: product).id`

Comment (medius) [inline api/spec/controllers/spree/api/products_controller_spec.rb]: Done.

**Comment by IrvinFan** [comment]
Looks good. Merging

**Pull request merged**